app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,75 @@
+class TweetController < ApplicationController
+	get "/tweets" do 
+	  	if !!session[:id]
+	  		erb :"tweets/tweets"
+	  	else
+	  		redirect "/login"
+	  	end
+  	end
+
+  	get "/tweets/new" do 
+  		if !!session[:id]
+  			erb :"tweets/create_tweet"
+  		else
+  			redirect "/login"
+  		end
+  	end
+
+  	post "/tweets" do 
+  		user = User.all.find(session["id"])
+		if params[:content] == ""
+			redirect "/tweets/new"
+		else
+			user.tweets << Tweet.create(params)
+			redirect "/tweets"
+		end
+  	end
+
+  	get "/tweets/:id" do 
+  		if !!session[:id]
+	  		@tweet = Tweet.find(params[:id])
+	  		erb :"tweets/show_tweet"
+	  	else
+	  		redirect "/login"
+	  	end
+  	end
+
+  	get "/tweets/:id/edit" do
+  		if !!session[:id]
+  			@tweet = Tweet.find(params[:id])
+  			erb :"tweets/edit_tweet"
+  		else
+  			redirect "/login"
+  		end
+  	end
+
+  	post "/tweets/:id" do
+  		tweet = Tweet.find(params[:id])
+  		if params[:content] == ""
+  			redirect "/tweets/#{tweet.id}/edit"
+  		else
+	  		tweet.content = params[:content]
+	  		tweet.save
+	  		redirect "/tweets/#{tweet.id}"
+	  	end
+  	end
+
+  	post "/tweets/:id/delete" do
+  		tweet = Tweet.find(params[:id])
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.